### ISSUE TITLE
Additional settings for Heating Circuit 2 and Summer time settings

### ISSUE THREAD

**Issue by kucerazdenek** [pail23/stiebel_eltron_isg_component#24]
@pail23 , excellent progress 0.8.1 !! Thanks a lot!

Would it be possible to add also settings for following parameters?

Heating Circuit 2

![image](https://user-images.githubusercontent.com/121700009/226173496-7f7de874-fb4e-4174-ab57-d93c00577806.png)

Summer mode
![image](https://user-images.githubusercontent.com/121700009/226173694-65332f0b-a66c-41cc-b99c-4902bf63ed58.png)

Summer time
![image](https://user-images.githubusercontent.com/121700009/226173901-ac003ff0-b060-490c-b46a-57d7c1aca3fa.png)

Thanks!

**Comment by pail23** [comment]
I did not find the corresponding registers in the modbus documentation: https://www.stiebel-eltron.ch/content/dam/ste/ch/de/downloads/kundenservice/smart-home/Modbus/Modbus%20Bedienungsanleitung.pdf

Do you know which ones to pick?

**Comment by xylone** [comment]
Bit weird that they are in de Service Welt, but not in the Modbus....
Same as the max temperature fields in the Energy dashboard:
<img width="962" alt="Scherm­afbeelding 2023-03-20 om 21 53 43" src="https://user-images.githubusercontent.com/2749350/226463503-c74b6cb4-d0bd-42ca-873a-cc97b4b150d7.png">

**Comment by kucerazdenek** [comment]
I raised this with local Stiebel support. During phone call they confirmed that should be possible to control those values via modbus registers, but asked for description in email and promised they will investigate.

Lets see with what they will come back.

**Comment by pail23** [comment]
Any news from the stiebel eltron support?

**Comment by kucerazdenek** [comment]
Unfortunately nothing, even I pushed them few times….

**Comment by kucerazdenek** [comment]
> Unfortunately nothing, even I pushed them few times…. 

One more question, I see that my ISG was updated to version 12.2.3. 
Any chance to get those two information from this?

![image](https://github.com/pail23/stiebel_eltron_isg_component/assets/121700009/3e10a87d-0d96-4ceb-beb6-f34755dc9d85)

**Comment by kucerazdenek** [comment]
Hi Paul, sorry for bothering, any chance to include at least those two values into integration?

Actual Compressor speed
Inverter Power

**Comment by pail23** [comment]
@kucerazdenek The "Verdichterstarts" (modbus register 31) is already implemented. This register is only available for LWZ7LWA heatpumps and not for WMP heat pumps.
Regarding the 361XX registers: Would it be possible for to test with the standard [modbus integration](https://www.home-assistant.io/integrations/modbus/) if you get the desired values? My heatpump does not provide those registers and therefore I can't test it. Once I get your confirmation that you can read those values and you get the correct content, I can implement it in the integration. Many thanks!

**Comment by kucerazdenek** [comment]
Hi @pail23 , sorry, have been stuck with some sad things last few months. Would you be able to give me some hint how to scan 361XX registers via modbus integration to test if requested values are provide through them?

**Comment by pail23** [comment]
Yes, you can used the home assistant [modbus integration](https://www.home-assistant.io/integrations/modbus/). There are also modbus apps available on mobile phones. Have a look in play store.

**Comment by kucerazdenek** [comment]
Sure, I tried below, but all trhee are shown as unavailable and for address 36118 (this one must be addressed in different way I am afraid), shows an error in HA log:
2024-03-07 06:44:08.799 ERROR (SyncWorker_3) [homeassistant.components.modbus.modbus] Pymodbus: modbus_test: Error: device: 1 address: 36118 -> pymodbus returned isError True

section in configuration.yaml

modbus:
  - name: "modbus_test"
    type: tcp
    host: 10.0.0.237
    port: 502

    sensors:
      - name: modbus_test_l1consumption
        address: 36118
        slave: 1
        scan_interval: 15
        unit_of_measurement: W
        data_type: float16
      - name: modbus_test_outside_temperature
        unit_of_measurement: °C
        address: 507
        slave: 1
        scan_interval: 15
        data_type: float16
      - name: modbus_test_actual_temperature_hk1
        unit_of_measurement: °C
        address: 508
        slave: 1
        scan_interval: 15
        data_type: float16

**Comment by Optic00** [comment]
A status update on the three things asked for here, since the integration has moved a long way since 2023.

**Heating circuit 2 is implemented.** Comfort and eco target temperature, heating curve rise and the low end of the curve exist as number entities for HK2, on the WPM controllers, and the same for HK3.

**Compressor starts and compressor speed are implemented** for LWZ controllers, which is where the underlying register exists, as `compressor_starts` and `compressor_speed`.

**Summer mode settings are not implemented.** The integration exposes summer mode only as the binary sensor `is_summer_mode`, so it reports the state but offers no entity to configure the threshold or the summer time, which is what the second and third screenshot ask for.

Leaving this open for that last part.

**Comment by Optic00** [comment]
Heating circuit 2 settings are now available. The compressor starts and speed values discussed later are also exposed for LWZ controllers where supported.

The remaining summer-mode and clock settings have no confirmed writable mapping in the register set used here. The integration currently reports summer mode as a read-only state.

I'm closing this issue as not planned for those remaining controls. If a documented write mapping becomes available, please open a follow-up for that setting.
